### ISSUE TITLE
More-polite GPS navigators

### ISSUE THREAD

**Issue by ItsNickBarry** [open-source-ideas/open-source-ideas#189]
## Project description

While the quality of GPS navigation voice synthesis has increased over the years, the courtesy of the navigators has not.  Not only do they interrupt conversations as if they're the only ones in the room, but they expect of their captive audience complete focus and comprehension and refuse to repeat themselves when asked.

It would be nice if the navigators would wait for at least the end of a sentence before speaking.  Perhaps an "ahem" would allow travellers to focus their attention on the forthcoming arbitrary information before it's too late.  As a last resort, one might ask, "navigator, please repeat yourself," and the navigator might oblige.

Software like this might even save lives, in addition to improving them, by preventing a few vehicular collisions.

## Relevant Technology

This would primarily be a mobile application (Android, iOS).  Audio analysis would be performed by AI.  Maps and directions might be provided by OpenStreetMaps.

### Complexity

- [ ] Beginner - This project requires no or little prior knowledge of the technolog(y|ies) specified to contribute to the project
- [ ] Intermediate - The user should have some prior knowledge of the technolog(y|ies) to the point where they know how to use it, but not necessarily all the nooks and crannies of the technology
- [x] Advanced - The project requires the user to have a good understanding of all components of the project to contribute

### Required time (ETA)

- [ ] Little work - A couple of days
- [ ] Medium work - A week or two
- [x] Much work - The project will take more than a couple of weeks and serious planning is required

### Categories

- [x] Mobile app
- [ ] IoT
- [ ] Web app
- [x] Frontend/UI
- [x] AI/ML
- [ ] APIs/Backend
- [x] Voice Assistant
- [ ] Developer Tooling
- [ ] Extension/Plugin/Add-On
- [ ] Design/UX
- [ ] AR/VR
- [ ] Bots
- [ ] Security
- [ ] Blockchain
- [ ] Futuristic Tech/Something Unique

**Comment by KOLANICH** [comment]
I don't think it is a good idea.

1. navigator app would have to listen. This means mic permission and privacy issues.
2. navigator app would have to detect if people are speaking and detect the smarker phrase for repeating. Means constantly doing inference on sound. It is a power consumption and heat dissipation issue.

3. and these all for what? A person driving a car should pay the most of attention to driving, not to talking to passengers. And remember that cars are fast, any delays would mean the driver will miss his turn and would have to either violate the traffic rules by stopping and turning driving backward or would have to make 2 U-turns.

**Comment by ItsNickBarry** [comment]
@KOLANICH I'm not convinced it's a good idea, either.  It's a minor UX improvement that existing providers should maybe consider, but probably doesn't justify building an entire navigation application.  However, to address your points:

1.  It's only a privacy concern if the audio is uploaded to a server.  This would not be necessary for the specified functions, and users could verify that there is no malicious code in the repository.
2.  I don't think battery usage is an issue because the phone can be charged while driving.   Audio processing would only need to be done for short intervals, just before and just after a direction is given.  Actually, I think battery usage could be manageable even if the phone is not plugged into a car: pedestrian journeys often have fewer course changes (for example, because one-way streets do not have to be avoided).  I think 40 seconds of audio processing over the course of a 30-minute trip would use very little battery when compared to the GPS itself over the same period.
3.  What drivers "should" do is irrelevant.  It is a fact that drivers talk to their passengers, and that is unlikely to change.  Yes, we should consider the ethical implications of any new software, but I don't think having this technology is incentive enough for conscientious drivers to adopt riskier behaviours.  Rather, the purpose is to prevent that risk that already exists from being compounded by GPS-related distraction (a distraction which may result in the driver's removing hands from the steering wheel or eyes from the road, changing lanes recklessly, or even driving backwards like you mentioned).  Also, this system would not necessarily result in delayed instructions; the first attempt to give instructions could simply be made 5-10 seconds earlier than usual, and if those seconds pass without a pause in conversation, the system would revert to default behaviour.

**Comment by KOLANICH** [comment]
>It's only a privacy concern if the audio is uploaded to a server. This would not be necessary for the specified functions, and users could verify that there is no malicious code in the repository.

1. One cannot make sure that the app corresponds to the source unless he has built it on a trusted machine. One will have to constantly verify app integrity after that.
2., 3. you are probably right.
<details>
<summary>offtop</summary>
It was not about ethical implications. It was about that it may not worth it. Some drivers speak actively. Some drivers are concentrated on driving. Most of drivers I know drive are not speaking constantly, so this feature won't be useful for them. Driving backwards is OK and even legal. It is illegal to pass the intersections this way.
</details>

**Comment by ludwigbald** [comment]
I agree that this problem exists. It has previously been adressed by:
* showing directions on screen
* The option to play only important commands (Google maps)
* Automatically turning down media volume
* Adding a "repeat last command" button on the tactile interface

I haven't seen a timing manipulation and I'm not sure how much good it would do.
Politeness is valued differently in different cultures. I suspect that here in Germany, people expect their navigation system to talk as little as possible. They would prefer clarity and deciciveness over polite phrases.

If anyone wants to give it a shot anyways, I suggest using maps.me as a base.
